routers/web/repo/editor.go

```diff
@@ -331,9 +354,7 @@ func editFilePost(ctx *context.Context, form forms.EditRepoFileForm, isNewFile b
 		_ = repo_model.UpdateRepositoryCols(ctx, &repo_model.Repository{ID: ctx.Repo.Repository.ID, IsEmpty: false}, "is_empty")
 	}
 
-	if form.CommitChoice == frmCommitChoiceNewBranch && ctx.Repo.Repository.UnitEnabled(ctx, unit.TypePullRequests) {
-		ctx.Redirect(ctx.Repo.RepoLink + "/compare/" + util.PathEscapeSegments(ctx.Repo.BranchName) + "..." + util.PathEscapeSegments(form.NewBranchName))
-	} else {
+	if !(form.CommitChoice == frmCommitChoiceNewBranch && redirectToPullRequest(ctx, branchName)) {
```

Comment:
I am not sure whether it is right here.

Before:

```
if .. {
   redirect()
} else {
   redirect()
}
```

now: 

```
if ! (... && tryToRedirect()) {
    redirect
}

```

1. it's difficult to understand & review the `tryToRedirect` part
2. it's not clear whether there is a "else" case which needs redirection (but it is missing?)

Reply:
Hopefully this is more clear?